src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/InteractionEndpointIT.java

```diff
@@ -0,0 +1,247 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.endpoint;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.CCStatus;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Case;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.CaseInteraction;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.CaseInteractionType;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.CollectionExercise;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Permission;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.PermissionType;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Role;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Survey;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.User;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CaseInteractionRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CaseRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CollectionExerciseRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.RoleRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.SurveyRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.UserAuditRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.UserRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.UsersCaseInteractionDTO;
+
+public class InteractionEndpointIT extends FullStackIntegrationTestBase {
+
+  private static final UUID TARGET_USER = UUID.fromString("4f27ee97-7ba7-4979-b9e8-bfe3063b41e8");
```

Comment:
would be better named 'principal' to keep in line with the terminology elsewhere in the code base.
The principal is always the user asking for the thing to be done, target makes it  sound as if something is being done to that user.